### PR TITLE
[0.19] Backports

### LINKED ISSUE #19510: Bitcoind not compiling on MacOS 10.15.5
I followed the instructions on `./docs/build-osx.md` but `make` command results in the following error:

```
qt/trafficgraphwidget.cpp:55:13: error: member access into incomplete type 'QPainterPath'
        path.moveTo(x, YMARGIN + h);
            ^
/usr/local/Cellar/qt/5.15.0/lib/QtGui.framework/Headers/qmatrix.h:54:7: note: forward declaration of 'QPainterPath'
class QPainterPath;
      ^
qt/trafficgraphwidget.cpp:59:17: error: member access into incomplete type 'QPainterPath'
            path.lineTo(x, y);
                ^
/usr/local/Cellar/qt/5.15.0/lib/QtGui.framework/Headers/qmatrix.h:54:7: note: forward declaration of 'QPainterPath'
class QPainterPath;
      ^
qt/trafficgraphwidget.cpp:61:13: error: member access into incomplete type 'QPainterPath'
        path.lineTo(x, YMARGIN + h);
            ^
/usr/local/Cellar/qt/5.15.0/lib/QtGui.framework/Headers/qmatrix.h:54:7: note: forward declaration of 'QPainterPath'
class QPainterPath;
      ^
qt/trafficgraphwidget.cpp:109:22: error: variable has incomplete type 'QPainterPath'
        QPainterPath p;
                     ^
/usr/local/Cellar/qt/5.15.0/lib/QtGui.framework/Headers/qmatrix.h:54:7: note: forward declaration of 'QPainterPath'
class QPainterPath;
      ^
qt/trafficgraphwidget.cpp:116:22: error: variable has incomplete type 'QPainterPath'
        QPainterPath p;
                     ^
/usr/local/Cellar/qt/5.15.0/lib/QtGui.framework/Headers/qmatrix.h:54:7: note: forward declaration of 'QPainterPath'
class QPainterPath;
      ^
5 errors generated.
make[2]: *** [qt/libbitcoinqt_a-trafficgraphwidget.o] Error 1
make[1]: *** [all-recursive] Error 1
```

### PR DESCRIPTION
**Pull request by fanquake**
Backports the following to the 0.19 branch:

* #17946 - Fix GBT: Restore "!segwit" and "csv" to "rules" key
* #18160 - gui: Avoid Wallet::GetBalance in WalletModel::pollBalanceChanged
* #18425 - releases: Update with new Windows code signing certificate
* #18676 - build: Check libevent minimum version in configure script
* #19097 - qt: Add missing QPainterPath include (as per #19510)
* #18640 - appveyor: Remove clcache
* #19444 - test: Remove cached directories and associated script blocks from appveyor config
* #19612 - lint: fix shellcheck URL in CI install
* #18001 -  Updated appveyor job to checkout a specific vcpkg commit ID

Closes: #19510.

### CODE REVIEW

**Comment by promag** [comment]
Do you want to include #18542 here?

**Comment by fanquake** [comment]
> Do you want to include #18542 here?

Thanks. I've pulled in d3cfe0d6a472c748142b96c33de866478cca5545 from that PR.

**Comment by luke-jr** [comment]
#19185 is a clean merge of the original commit. No need for an ugly re-commit as here.

Should remove the commits here, and reopen #19185 instead

**Review by luke-jr — CHANGES_REQUESTED** [review]
#19185 is a clean merge of the original commit. No need for an ugly re-commit as here.

Should remove the commits here, and reopen #19185 instead

**Comment by promag** [comment]
#19300 was merged.

**Comment by jnewbery** [comment]
ACK be951470bea132b4cbe1823cd564bf14aaf0ea7e

**Comment by MarcoFalke** [comment]
cherry-pick ACK be951470bea132b4cbe1823cd564bf14aaf0ea7e 🌎

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

cherry-pick ACK be951470bea132b4cbe1823cd564bf14aaf0ea7e 🌎
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhiqAv+LLwsV4mNGsaTSOa3q9mT14B/bWuib9+CFffg8lBvedHfjs/B/jRRkbDt
R8zCFOZPJHed/8M3ICUYHi04FJkO5y4Bh8TzIY8087SPM6sDatE/MWXSCe5f9Ttt
aFTfYx19As/yaE4cMJZD+/3pwHVObZuQ4nzu9TZYbtUBRoJj912QP5Yq+gBZXxwt
M10O6pwz7e5IXwGCPeIIlhMLgiqmom7pJIsijZi3PaGeH+oR8sBnxB99B4dVwiJl
+E7VQSVldHNjoNcSftvp41deDc1L79hFn3au4ws2Zq6y1BXlgAxLOqBmkGSS6y1H
Qvq+YrtQM/pGblgRuXUD3Nb3qbehk7mVPka+RlI3diyj1yB6YbG0YAjDxvDmkQgp
VBqW7loKrAnxO6VRpaZ8+SGZyWwCQQQEii25WKvRZm4xR/5AoJgHv4PU1hK89jei
lzayyDZVUqbWS8PDD4zhes28m1u5iZaTKgALYFegyxnRl5YHuwSfACuLjQT2CMYM
wpMzwC5O
=AqoC
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `85457e8568a50d11680719483b91ec171c3e108883edca3d38bbae169190b69a  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010885457e8568a50d11680719483b91ec171c3e108883edca3d38bbae169190b69af0108007f04033f96fa8ccd06c1c66fd39ca08fff01031cdb7efbba19611fa452b7321e3b83608f1045f325fbff008fb74f240498991840083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010b8be27e7bc024f17e93377cc6df037ec08f1045f325fbef0084dd6694a334568d00083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010cc9e343415e3f0a88fa1784e1540b1ec08f1045f325fbef00885be8dc39a313b990083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

File: .appveyor.yml [hunk 1]
```diff
@@ -5,78 +5,54 @@ configuration: Release
...
-      }
-      else {
-        Write-Host "required vcpkg packages already installed."
-      }
-      c:\tools\vcpkg\vcpkg integrate install
+      Write-Host "vcpkg installing packages: $env:PACKAGES"
+      cd c:\tools\vcpkg
+      $env:GIT_REDIRECT_STDERR = '2>&1' # git is writing non-errors to STDERR when doing git pull. Send to STDOUT instead.
+      git pull origin master > $null
+      git checkout $env:VCPKG_COMMIT_ID
```
Comment (MarcoFalke) [inline .appveyor.yml:27]: Any reason to check out the same commit twice?

You can avoid the conflicts by cherry-picking in the correct order:

```
$ git log -3 --oneline 
a51ffbe92c (HEAD) Remove cached directories and associated script blocks from appveyor CI configuration.
6646b5f891 appveyor: Remove clcache
bd95892b0d Updated appveyor job to checkout a specific vcpkg commit ID.

**Comment by MarcoFalke** [comment]
> #19300 was merged.

This wasn't included here, but I went ahead and merged this to make the ci green on the other backport pulls

**Pull request merged**